### DISCUSSION TITLE
Animated Transfrom from one Icon to Another

### DISCUSSION THREAD

**Discussion by DarkSuniuM** [tailwindlabs/tailwindcss#4524] [Help]
Hello,
I'm trying to create a floating button which has 2 states, Open and Close, On each of these states, I use a different Icon, 

I'm using React so the minimal version of my code would be something like this:

```javascript
import { useState, useMemo } from "react";
import { MenuIcon, XIcon } from "@heroicons/react/solid";

function FloatingButton() {
  const [isOpen, setIsOpen] = useState(false);
  const Icon = useMemo(() => isOpen ? XIcon : MenuIcon, [isOpen]);
  return (
    <button
      onClick={() => setIsOpen(prev => !prev)}
      className="bg-pink-500 fixed right-2 bottom-2 md:hidden rounded-full h-10 w-10 flex items-center justify-center z-40"
    >
      <Icon className="p-2" />
    </button>
  );
}
```

I'm trying to add an animation on the `onClick` action of my button, a transform from `XIcon` to `MenuIcon` and `MenuIcon` to `XIcon` based on the current state of the component.

I do understand why my approach does not work, but I have no clue what approach would work.

I already tried `react-svg-morph` and it's 2 component (`MorphReplace`, and `MorphTransition`), resulted in not showing the Icons.
Tried to have animation with conditional classNames like `(isOpen ? " hidden" : "")`, didn't work either.

Is there anyway to apply such a behavior using Tailwind?

Thank you all in advance :pray:

**Comment by devntv** [comment]
sorry I don't understand your problem. Can you explain in more detail?

**Reply by DarkSuniuM** [reply]
Hello @devntv 
Of course
Take a look at this => https://codepen.io/designcouch/pen/Atyop

Click on the Icons, they have animations, an SVG is tranfromed to another (burger to close, close to burger)
I wanted to implement something like this with Tailwind and heroIcons

**Reply by devntv** [reply]
> Hello @devntv
> Of course
> Take a look at this => https://codepen.io/designcouch/pen/Atyop
> 
> Click on the Icons, they have animations, an SVG is tranfromed to another (burger to close, close to burger)
> I wanted to implement something like this with Tailwind and heroIcons

i think i can help you with this with `useState` + `tailwind` +` heroIcon`
first, i think it might be the same as yours
(you make your own animation)
first just add your animation(custom) in tailwind.config.js  file
``` 
theme : {
       extend: {
         keyframes: {
             yourAnimtionCustom: {
                  '0%': {transform: 'scale(1.3)', opacity: '0'},
                  '65%': {opacity: '.8'},
                  '100%': {transform: 'scale(1)',opacity: '1'}
             }
         }
            animation: { yourAnimtionCustom : 'yourAnimtionCustom .3s ease-out'}
    }
}
// in component that you want handle
   const [toggle, setToggle] = useState(false)
   const handleToggle = () => setToggle(!toggle)
// heroIcon svg - animation on 1 svg
<svg 
  onClick={handleToggle} 
  className={`h-8 w-8 ${toggle ? 'animate-yourAnimtionCustom':'className,animation that you want it to apply here'}`}
  >path='......'</svg>

**Reply by devntv** [reply]
you have to make your own animation
  I'm really sorry I coded it by hand, not with vscode so there might be errors. You just reply, if i can handle it i will try to help you
------------------------------------------------------

// heroIcon svg - transform animation 2 svg
  ```
{ toggle ? <svg1 onClick={handleToggle} className='h6 w-6 animate-yourAnimtionCustom'>path='...'</svg1>
              : <svg2 onClick={handleToggle} className='h6 w-6 animate-yourAnimtionCustom2'>path='....'</svg2>
  }
